docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +165,166 @@ Note that:
+    { path = 'mqtt://configs/bar.conf', type = 'bar.conf', childid = 'child1'  }
+  ]
+```
+
+Example Flow:
+
+**Start Behaviour:**
+  * external device `child1`: starts
+  * external device `child1`: publishs all its config files to MQTT (with retain); Example: Publish to topic `configs/bar.conf`
+
+**Device-to-Cloud Behaviour:**
+  * at some point a config retrieval for type `bar.conf` for `child1` arrives at C8Y config plugin
+  * C8Y config plugin: subscribes to `configs/bar.conf` and gets immediately the config file content from the retained message
+  * C8Y config plugin: sends the recived MQTT payload as config file to C8Y<br/><br/>
+    NOTE: The responsibility to assure that the latest config file content is on the MQTT bus is always on the process/external device.
+When there is no retain message the plugin sends an error to the cloud on an every incoming config retrieval request.
+
+**Cloud-to-Device Behaviour:**
+  * at some point a config sent from cloud for type `bar.conf` for `child1` arrives at C8Y config plugin
+  * C8Y config plugin: publishs the config file content as MQTT retain message to `configs/bar.conf` 
+  * external device `child1`: recognizes the MQTT message on `configs/bar.conf` and processes the new config file content
+
+--------------------------------------------------------------------------------
+
+## Declaration of Capabilities for Child-Devices
+
+The `c8y_configuration_plugin` takes care to define all necessary capabilities to each cloud's child-device twin. These are:
+  - declaring _supported operations_ for configuration management: `c8y_UploadConfigFile` and `c8y_DownloadConfigFile`
+  - declaring provided _configuration types_
+
+**Declaring 'supported operations'**
+
+For each `childid` contained in the plugin's configuration, the plugin sends one MQTT message to C8Y. If one `childid` occurs more than once just one message will be sent. The message is as below:
+  - topic: `c8y/s/us/<childid>`
+  - payload: `114,c8y_UploadConfigFile,c8y_DownloadConfigFile`
+
+
+**Declaring 'provided configuration types'**
+
+For each `childid` contained in the plugin's configuration, the plugin sends one MQTT message to C8Y. Thereby all `config types` assigned to that `childid` will be combined in that single message. The message is as below:
+  - topic: `c8y/s/us/<childid>`
+  - payload: `119,<config type 1>,<config type 2>,<config type 3>,...`
+
+
+**When to send capability messages**
+
+The `c8y_configuration_plugin` sends both MQTT message from above (for _supported operations_ and _provided configuration types_), always when the plugin's configuration is (re)processed. That means:
+  - on startup of the plugin
+  - always when the plugin recognizes a configuration change and processes it's configuration again
+
+
+Note that the `c8y_configuration_plugin` does **not** create any child-device twin in the cloud. Instead the clouds child-device twins must be created upfront.
```

Comment:
Good point. My intention here was to avoid spending effort in implementing something complex (i.E. checking for and creating the device-twin), that we might throw away later. That is since we have not yet confirmed about the general way and responsibilities of child-device provisioning (among others the question: "**Who** creates **when** the device-twin?").

On the other hand when we implement nothing here, a device twin will be created by C8Y automatically, once thin-edge sends the supported operations SmartREST message `114,c8y_UploadConfigFile,c8y_DownloadConfigFile` to the cloud. And as we know that child-device twin is then not in a proper shape: 
   - it's _Name_ has the prefix "MQTT Device "
   - it's _Type_ is "c8y_MQTTChildDevice", 
   instead of "thin-edge.io-child" - as thin-edge puts for child-devices created by measurements or events.

So that automatically created device-twin could be misleading for one who did not create the child-device twin upfront.

However that  _child-device provisioning_ concept need anyway to be defined and confirmed as a very next step. I will incorporate the outcome here as soon as there is a decision.

Reply:
Understood. Then, in my opinion, it would be better staying as "error" unless we have the further specs & implementation of child device creation mechanism.